src/main/java/org/apache/commons/validator/routines/checkdigit/VATidFRCheckDigit.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.validator.routines.checkdigit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.validator.GenericTypeValidator;
+import org.apache.commons.validator.GenericValidator;
+import org.apache.commons.validator.routines.SireneValidator;
+
+/**
+ * French VAT identification number (VATIN) Check Digit calculation/validation.
+ * <p>
+ * le numéro d’identification à la taxe sur la valeur ajoutée (no TVA)
+ * </p>
+ * <p>
+ * See <a href="https://en.wikipedia.org/wiki/VAT_identification_number">Wikipedia - VAT IN</a>
+ * for more details.
+ * </p>
+ *
+ * @since 1.10.0
+ */
+public final class VATidFRCheckDigit extends ModulusCheckDigit {
+
+    private static final long serialVersionUID = -4684483629166798828L;
+    private static final Log LOG = LogFactory.getLog(VATidFRCheckDigit.class);
+
+    /** Singleton Check Digit instance */
+    private static final VATidFRCheckDigit INSTANCE = new VATidFRCheckDigit();
+
+    /**
+     * Gets the singleton instance of this validator.
+     * @return A singleton instance of the class.
+     */
+    public static CheckDigit getInstance() {
+        return INSTANCE;
+    }
+
+    static final int LEN = 11; // with Check Digit
+    static final int CHECKDIGIT_LEN = 2;
+
+    private static final String ALPHABET = "0123456789ABCDEFGHJKLMNPQRSTUVWXYZ";
+    private static final int MODULUS_97 = 97;
+    private static final SireneValidator SIRENE_VALIDATOR = SireneValidator.getInstance();
+
+    /**
+     * Constructs a Check Digit routine.
```

Comment:
Not helpful IMO. Also, there is no "routine" concept in Java, this constructs a new instance if anything ;-)